packages/types/src/codec/EnumType.spec.js

```diff
@@ -33,4 +33,15 @@ describe('Struct', () => {
       ).toString()
     ).toEqual('Text');
   });
+
+  it('allows accessing the type and value', () => {
+    const text = new Text('foo');
+    const enumType = new EnumType(
+      [Text, U32],
+      { Text: text }
+    );
+
+    expect(enumType.type).toBe('Text');
+    expect(enumType.value).toEqual(text);
```

Comment:
@jacogr I added a test here, you should be able to access the .raw value like this. Why it returns a Base on the trie is a big unknown for me, I can have a look once it's more stabilized.